townhall/myapi/dao.py

```diff
@@ -251,3 +253,44 @@ def filtered_organization(
             filters["location__icontains"] = filtered_organization_data.location
 
         return Organization.objects.filter(**filters)
+
+class TaskDao:
+
+    @staticmethod
+    def get_all_tasks():
+
+        return Task.objects.all()
+
+    @staticmethod
+    def get_task_by_id(task_id: int):
+
+        try:
+            return Task.objects.get(id=task_id)
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def create_task(task_data: dict):
+
+        return Task.objects.create(**task_data)
+
+    @staticmethod
+    def update_task(task_id: int, task_data: dict):
```

Comment:
I get the idea here with the dict, but this needs validation and/or structure such that someone doesnt try to pass in keys that dont make sense such as fields that dont exist. The use of dataclass to define this task_data param will be useful here. https://www.datacamp.com/tutorial/python-data-classes 

I see in types.py some have been added for the create operation, similar thing can be done here

Reply:
Made it more robust